presidio-anonymizer/app.py

```diff
@@ -1,21 +1,27 @@
-"""TODO: Flask application instantiation and route definition."""
-
 from flask import Flask, request
 
+from presidio_anonymizer.anonymizer_engine import AnonymizerEngine
+from presidio_anonymizer.entities.engine_request import AnonymizerEngineRequest
+from presidio_anonymizer.entities.invalid_exception import InvalidParamException
+
 app = Flask(__name__)
 
 
-# TODO add method for get supported anonymizers. 2629
 @app.route("/anonymize", methods=["POST"])
 def anonymize():
-    """
-    TODO: Anonymize endpoint definition.
-
-    :return:
-    """
-    content = request.json
-    return content
+    content = request.get_json()
+    if not content:
+        return "Invalid request json", 400
+    try:
+        data = AnonymizerEngineRequest(content)
+    except InvalidParamException as e:
+        return e.err, 400
+    except Exception:
+        # TODO add logger 2652
+        return "Internal server error", 500
```

Comment:
Since this is an API it is a bad practice to return internal errors.
It is ok if the user is using the lib itself (the engine). 
So it is better to mask them with 500 internal error and log them on the server side.

Reply:
1. Can it be done in a global exception handler then? what if we add another endpoint and someone forgets to add this? 
2. Below, you call 'text = AnonymizerEngine().anonymize(data)'. If it will fail, you aren't masking it right?

Reply:
sure.